draft-gpew-priv-ppm.md

```diff
@@ -947,26 +938,30 @@ aggregator has validated and which fall in the batch interval of the request.
 
 Using a client-provided report multiple times within a single batch, or using
 the same report in multiple batches, may allow a server to learn information
-about the client's measurement, violating the privacy property of PPM. To
-prevent such replay attacks, this specification defines a total ordering on
-reports that aggregators can use to ensure that reports are aggregated once.
-
-Aggregate requests are ordered as follows: We say that a report `R2` follows
-report `R1` if either `R2.time > R1.time` or `R2.time == R1.time` and `R2.nonce
-> R1.nonce`. If `R2.time < R1.time`, or `R2.time == R1.time` but `R2.nonce <=
-R1.nonce`, then we say that `R2` does not follow `R1`.
-
-To prevent replay attacks, each aggregator ensures that each report it
-aggregates follows the previously aggregated report. To prevent the adversary
-from tampering with the ordering of reports, honest clients incorporate the
-ordering-sensitive parameters `(time, nonce)` into the AAD for HPKE encryption.
-Note that this strategy may result in dropping reports that happen to have the
-same timestamp and nonce value.
+about the client's measurement, violating the privacy goal of PPM. To
+prevent such replay attacks, this specification requires the aggregators to
+detect and filter out replayed reports.
+
+To detect replay attacks, each aggregator keeps track of the set of nonces
+pertaining to reports that were previously processed for a given task. If the
+leader receives a report from a client whose nonce is in this set, it simply
+ignores it. Similarly, a helper ignores an encrypted input share whose nonce is
+in this set.
+
+To prevent the adversary from tampering with the ordering of reports, honest
+clients incorporate the nonce into the AAD for HPKE encryption. Note that this
+strategy may result in dropping reports that happen to have the same nonce.
```

Comment:
This note being attached to the AAD statement is confusing.  It's a note about the whole scheme, not the AAD construction.

Reply:
Good call. I replaced this with a paragraph that, hopefully, makes it more clear what we're trying to accomplish here.